### DISCUSSION TITLE
Get the same result for my forge test

### DISCUSSION THREAD

**Discussion by QsAdedipe** [Cyfrin/foundry-full-course-cu#2294] [Q&A]
Pls, why am I getting the same result for two different forge test? below is my test code:

"  modifier funded() {
        vm.prank(USER);
        fundMe.fund{value: sentValue}();
        _;
    }

    function testOnlyOwnerCanWithdraw() public funded {
        //vm.prank(expectedFunder);
        //vm.prank(USER);
        vm.expectRevert();
        vm.prank(USER);

        fundMe.withdraw();
    }
}
"
This is the code function:

" function withdraw() public onlyOwner {
        for (uint256 funderIndex=0; funderIndex < s_funders.length; funderIndex++){
            address funder = s_funders[funderIndex];
            s_addressToAmountFunded[funder] = 0;
        }
        s_funders = new address[](0);
        // // transfer
        // payable(msg.sender).transfer(address(this).balance);
        
        // // send
        // bool sendSuccess = payable(msg.sender).send(address(this).balance);
        // require(sendSuccess, "Send failed");

        // call
        (bool callSuccess, ) = payable(msg.sender).call{value: address(this).balance}("");
        require(callSuccess, "Call failed");
    }"

**Comment by EngrPips** [comment]
```js
 modifier funded() {
vm.prank(USER);
fundMe.fund{value: sentValue}();
_;
}

function testOnlyOwnerCanWithdraw() public funded {
    //vm.prank(expectedFunder);
    //vm.prank(USER);
    vm.expectRevert();
    vm.prank(USER);

    fundMe.withdraw();
}
}
```

```js
function withdraw() public onlyOwner {
for (uint256 funderIndex=0; funderIndex < s_funders.length; funderIndex++){
address funder = s_funders[funderIndex];
s_addressToAmountFunded[funder] = 0;
}
s_funders = new address;
// // transfer
// payable(msg.sender).transfer(address(this).balance);

    // // send
    // bool sendSuccess = payable(msg.sender).send(address(this).balance);
    // require(sendSuccess, "Send failed");

    // call
    (bool callSuccess, ) = payable(msg.sender).call{value: address(this).balance}("");
    require(callSuccess, "Call failed");
}
```

**Comment by EngrPips** [comment]
Hello @QsAdedipe, You said you are getting the same result for two tests but you only posted one test. Where is the other test?

**Comment by QsAdedipe** [comment]
modifier funded() {
        vm.prank(USER);
        fundMe.fund{value: sentValue}();
        _;
    }

    function testOnlyOwnerCanWithdraw() public funded {
        vm.prank(expectedFunder);
        //vm.prank(USER);
        vm.expectRevert();
       // vm.prank(USER);

        fundMe.withdraw();
    }

On Thu, Aug 22, 2024 at 1:53 PM EngrPips ***@***.***> wrote:

> Hello @QsAdedipe <https://github.com/QsAdedipe>, You said you are getting
> the same result for two tests but you only posted one test. Where is the
> other test?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Cyfrin/foundry-full-course-cu/discussions/2294#discussioncomment-10419067>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXKXH2M43H4X4WQFXFXYF6LZSXNMDAVCNFSM6AAAAABM6B7436VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBRHEYDMNY>
> .
> You are receiving this because you were mentioned.Message ID:
> <Cyfrin/foundry-full-course-cu/repo-discussions/2294/comments/10419067@
> github.com>
>


-- 
QS Adedipe Oluwasina, MNIQS, RQS

**Comment by QsAdedipe** [comment]
it is "USER" that is funding the tx and only the funder can call the
withdrawal function, so I was testing if the expectedFunder can also call
the withdrawal function without being the funder


On Thu, Aug 22, 2024 at 1:58 PM OLUWASINA Adedipe ***@***.***>
wrote:

>  modifier funded() {
>         vm.prank(USER);
>         fundMe.fund{value: sentValue}();
>         _;
>     }
>
>     function testOnlyOwnerCanWithdraw() public funded {
>         vm.prank(expectedFunder);
>         //vm.prank(USER);
>         vm.expectRevert();
>        // vm.prank(USER);
>
>         fundMe.withdraw();
>     }
>
> On Thu, Aug 22, 2024 at 1:53 PM EngrPips ***@***.***> wrote:
>
>> Hello @QsAdedipe <https://github.com/QsAdedipe>, You said you are
>> getting the same result for two tests but you only posted one test. Where
>> is the other test?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Cyfrin/foundry-full-course-cu/discussions/2294#discussioncomment-10419067>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AXKXH2M43H4X4WQFXFXYF6LZSXNMDAVCNFSM6AAAAABM6B7436VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBRHEYDMNY>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> <Cyfrin/foundry-full-course-cu/repo-discussions/2294/comments/10419067@
>> github.com>
>>
>
>
> --
> QS Adedipe Oluwasina, MNIQS, RQS
>


-- 
QS Adedipe Oluwasina, MNIQS, RQS

**Reply by EngrPips** [reply]
Yeah, you will always get the same result no matter what address you prank to call the `withdraw` function as it is only the owner of the `FundMe` contract that can call the `withdraw` function.

**Comment by QsAdedipe** [comment]
Ok, thanks for the clarification.

On Thu, Aug 22, 2024 at 2:03 PM EngrPips ***@***.***> wrote:

> Yeah, you will always get the same result no matter what address you prank
> to call the withdraw function as it is only the owner of the FundMe
> contract that can call the withdraw function.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Cyfrin/foundry-full-course-cu/discussions/2294#discussioncomment-10419190>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXKXH2P5FMMFAXQ5YK3HL6LZSXORBAVCNFSM6AAAAABM6B7436VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBRHEYTSMA>
> .
> You are receiving this because you were mentioned.Message ID:
> <Cyfrin/foundry-full-course-cu/repo-discussions/2294/comments/10419190@
> github.com>
>


-- 
QS Adedipe Oluwasina, MNIQS, RQS

**Reply by EngrPips** [reply]
You are welcome. Keep crushing it. Please feel free to mark the discussion as answered as well.

**Comment by QsAdedipe** [comment]
also pls, My test result when i deploy on ETHSepolia is different from the
one of anvi for my expectRevert vm.

On Thu, Aug 22, 2024 at 2:04 PM OLUWASINA Adedipe ***@***.***>
wrote:

> Ok, thanks for the clarification.
>
> On Thu, Aug 22, 2024 at 2:03 PM EngrPips ***@***.***> wrote:
>
>> Yeah, you will always get the same result no matter what address you
>> prank to call the withdraw function as it is only the owner of the FundMe
>> contract that can call the withdraw function.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Cyfrin/foundry-full-course-cu/discussions/2294#discussioncomment-10419190>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AXKXH2P5FMMFAXQ5YK3HL6LZSXORBAVCNFSM6AAAAABM6B7436VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBRHEYTSMA>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> <Cyfrin/foundry-full-course-cu/repo-discussions/2294/comments/10419190@
>> github.com>
>>
>
>
> --
> QS Adedipe Oluwasina, MNIQS, RQS
>


-- 
QS Adedipe Oluwasina, MNIQS, RQS

**Reply by EngrPips** [reply]
What is the difference?

**Comment by QsAdedipe** [comment]
This is my test
" function testFundFailsWithoutEnoughETH() public {
        vm.expectRevert(); //the next line should revert
        fundMe.fund{value: 10e10}();
    }"
 and this is the function I am testing
"contract FundMe {
    using PriceConverter for uint256;

    mapping(address => uint256) private s_addressToAmountFunded;
    address[] public s_funders;

    // Could we make this constant?  /* hint: no! We should make it
immutable! */
    address public /* immutable */ i_owner;
    uint256 public constant MINIMUM_USD = 5 * 10 ** 18;
    AggregatorV3Interface private s_priceFeed;

    constructor(address priceFeed) {
        i_owner = msg.sender;
        s_priceFeed = AggregatorV3Interface(priceFeed);
    }

    function fund() public payable {
        require(msg.value.getConversionRate(s_priceFeed) >= MINIMUM_USD, "You
need to spend more ETH!");
        // require(PriceConverter.getConversionRate(msg.value) >=
MINIMUM_USD, "You need to spend more ETH!");
        s_addressToAmountFunded[msg.sender] += msg.value;
        s_funders.push(msg.sender);
    }"
when deployed on anvi chain my test passed but when deployed on sepolia it
passed. why is this happening?
On Thu, Aug 22, 2024 at 2:37 PM EngrPips ***@***.***> wrote:

> What is the difference?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Cyfrin/foundry-full-course-cu/discussions/2294#discussioncomment-10419588>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXKXH2KBW5BXYMK5DYAVMHLZSXSSTAVCNFSM6AAAAABM6B7436VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBRHE2TQOA>
> .
> You are receiving this because you were mentioned.Message ID:
> <Cyfrin/foundry-full-course-cu/repo-discussions/2294/comments/10419588@
> github.com>
>


-- 
QS Adedipe Oluwasina, MNIQS, RQS

**Reply by EngrPips** [reply]
Hello @QsAdedipe, Do you mean your test is failing on Sepolia, If so what is the test failing with?

**Reply by QsAdedipe** [reply]
I am getting different test results

**Reply by EngrPips** [reply]
You were not supposed to get a different result?

**Reply by EngrPips** [reply]
What result were you supposed to get, and what did you get?
